File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        
        int len=nums.size();
        vector<int>res;
        for(int i=0;i<len;i++){
            if(nums[abs(nums[i])-1]>0)
                nums[abs(nums[i])-1]*=-1;
            else
                res.push_back(abs(nums[i]));
        }
        
        for(int i=0;i<len;i++){
            if(nums[i]>0)
                res.push_back(i+1);
        }
        return res;
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        
        
        int len=nums.length;
        
        int sum=0;
        
        for(int i:nums){
            sum+=i;
        }
        
        for(int i=0;i<len;i++){
            int abs=Math.abs(nums[i]);
            
            if(nums[abs-1]<0){
                return new int[]{abs,abs+(len*(len+1))/2-sum};
            }
            
            nums[abs-1]*=-1;
        }
        
        return new int[0];
    }
}